src/System.Threading.Tasks.Channels/System/Threading/Tasks/Channels/WritableChannel.cs

```diff
@@ -18,9 +18,9 @@ public abstract class WritableChannel<T>
         /// <param name="error">An <see cref="Exception"/> indicating the failure causing no more data to be written, or null for success.</param>
         /// <returns>
         /// true if this operation successfully completes the channel; otherwise, false if the channel could not be marked for completion,
-        /// for example due to having already been marked as such.
+        /// for example due to having already been marked as such, or due to not supporting completion.
         /// </returns>
-        public abstract bool TryComplete(Exception error = null);
+        public virtual bool TryComplete(Exception error = null) => false;
```

Comment:
Why this change?

Reply:
It's essentially saying that the default is an infinite channel; Completion returns a never-completing task by default, and TryComplete returns false to say you can't complete it.  If a derived channel wants to override it, it of course can, and do so in a way that appropriately coordinates between the read side and the write side.  We could go back to having them both be abstract, but it seemed like an unnecessary additional thing to be forced to override in order to implement a custom channel.

I don't feel strongly about it, though.  Happy to change it back if you disagree.

Reply:
Ah, thanks. The change makes sense.